events-application-rest-api/src/main/java/com/bounswe/bounswe2017group3/UserController.java

```diff
@@ -39,11 +43,36 @@ public UserController(UserRepository repository) {
         this.repository = repository;
     }
 
+    @RequestMapping(method = RequestMethod.GET, value = "",
+            params="username",
+        produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public @ResponseBody User userByUsername(@RequestParam("username") String username){
+        return repository.findByUsername(username);
+    }
+    
     @RequestMapping(method = RequestMethod.GET,
         produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
     public @ResponseBody List<User> user(ModelMap model) {
         return repository.findAll();
     }
+     
+    @RequestMapping(method=RequestMethod.PUT,
+            value="",
+            params="username",
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public @ResponseBody User update(@RequestParam MultiValueMap<String, String> params){
+        
+        String oldUsername = params.get("username").get(0);
+        String newUsername = params.get("username").get(1);
+        User updatedUser = repository.findByUsername(oldUsername);
+        updatedUser.setEmail(params.get("email").get(0));
+        updatedUser.setFullname(params.get("fullname").get(0));
+        updatedUser.setUsername(newUsername);
+        repository.save(updatedUser);
```

Comment:
Maybe we should put some error handling here.
- If the user data is invalid, we can return an error message (in a JSON object) with the HTTP status code 400 (Bad Request).
- If the server somehow failed to update the data (but the input is valid), we can return an error message (in a JSON object) with the HTTP status code 500 (Internal Server Error).

For example, user can give someone else's email address and the DB would fail to save the new data. In this case, the input is invalid. We can display an error like this:
```json
{
  "error": {
    "code": 1013,
    "message": "Given email address belongs to someone else."
  }
}
```
I just made up the 1013 error code and the error message, but you get the point 😄 Note that 1013 error code is **different from 400**. 1013 is our custom code here, which we'll have to document in our API documentation, while 400 is the status code we give to the user as a response to the HTTP request.